builder.nims

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env nim
+
+## This file is a prototype of what should be our new API for the CLI builder
+## The purpose is to have a better, readable and extensible script for the 
+## future implementations.
+## 
+## Since Artuto may grow or even change dependencies, support or update them,
+## make this code easier to change is our best choice.
+##
+## 
+## This file depends on ``tools/builderUtils.nims`` that makes the dirty job.
+
+import "./tools/builderUtils.nims"
+
+cmd build, "Builds Arturo":
+    ## build:
+    ##     Provides a cross-compilation for the Arturo's binary.
+    ## 
+    ##     --arch: string = $hostCPU
+    ##          [amd64, arm, arm64, i386, x86]
+    ##     --build -b: string = full
+    ##          [full, mini, web]
+    ##     --os: string = $hostOS
+    ##          [freebsd, linux, openbsd, macosx, netbsd, windows]
+    ##     --who: string = user
+    ##          [bench, ci, dev, user]
+    ##     --compress -c
+    ##     --install -i
+    ##     --log -l
+    ##     --help
+
+    const
+        availableOSes  = @["freebsd", "linux", "openbsd", "macosx", "netbsd", "windows"]
+        availableCPUs  = @["amd64", "arm", "arm64", "i386", "x86"]
```

Comment:
- Idea: add `x86-64` and `x64` as `amd64` alias.

> In 1999–2003, [AMD](https://en.wikipedia.org/wiki/Advanced_Micro_Devices) extended this 32-bit architecture to 64 bits and referred to it as [x86-64](https://en.wikipedia.org/wiki/X86-64) in early documents and later as [AMD64](https://en.wikipedia.org/wiki/AMD64). Intel soon adopted AMD's architectural extensions under the name IA-32e, later using the name EM64T and finally using Intel 64. [Microsoft](https://en.wikipedia.org/wiki/Microsoft) and [Sun Microsystems](https://en.wikipedia.org/wiki/Sun_Microsystems)/[Oracle](https://en.wikipedia.org/wiki/Oracle_Corporation) also use term "x64", while many [Linux distributions](https://en.wikipedia.org/wiki/Linux_distribution), and the [BSDs](https://en.wikipedia.org/wiki/BSD) also use the "amd64" term. Microsoft Windows, for example, designates its 32-bit versions as "x86" and 64-bit versions as "x64", while installation files of 64-bit Windows versions are required to be placed into a directory called "AMD64".[[21]](https://en.wikipedia.org/wiki/X86#cite_note-32)
> - Wikipedia

source: https://en.wikipedia.org/wiki/X86